ios/brave-ios/Sources/Brave/Frontend/Browser/TabManager.swift

```diff
@@ -906,13 +906,15 @@ class TabManager: NSObject {
     }
 
     // Remove all unwanted tabs
-    for tab in allTabs {
-      guard tab.visibleURL?.urlToShred?.baseDomain == baseDomain else { continue }
+    for tabToClose in allTabs {
+      guard tabToClose.visibleURL?.urlToShred?.baseDomain == baseDomain,
+        tabToClose.isPrivate == tab.isPrivate
+      else { continue }
```

Comment:
nit: May be better to filter before iterating or use the `where` syntax (`for tabToClose in allTabs where ...`)